### PR TITLE
Add support for provisioning updates via dell-hwmgr

### PR DESCRIPTION
**Pull request by donpenney**

### CODE REVIEW

**Review by rauhersu — APPROVED** [review]

File: internal/controller/o2ims-hardwaremanagement/nodepool_controller.go [hunk 1]
```diff
@@ -113,6 +113,15 @@ func (r *NodePoolReconciler) Reconcile(ctx context.Context, req ctrl.Request) (r
 
 // SetupWithManager sets up the controller with the Manager.
 func (r *NodePoolReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	// Setup Node CRD indexer
```
[the remark this answers is not in the document]
Comment (donpenney) [reply internal/controller/o2ims-hardwaremanagement/nodepool_controller.go:116]: I can add a comment to this. It was needed in order to do a filtered List query using a field in the spec.

File: adaptors/dell-hwmgr/nodepool.go [hunk 2]
```diff
@@ -283,143 +274,145 @@ func (a *Adaptor) ReleaseNodePool(ctx context.Context,
...
-	}
+		// Set the new profile in the spec
+		node.Spec.HwProfile = newHwProfile
 
-	// Stage 1: Initiate upgrades by updating node.Spec.HwProfile as necessary
-	for _, name := range allocatedNodes {
-		node, err := utils.GetNode(ctx, a.Client, a.Namespace, name)
-		if err != nil {
-			return utils.RequeueWithShortInterval(), err
+		// Record the jobId in an annotation
```
[the remark this answers is not in the document]
Comment (donpenney) [reply adaptors/dell-hwmgr/nodepool.go:391]: Yeah, that's been in the back of my mind to do for both this and the initial resource group creation. Once the job is completed successfully, I don't think there's a need to keep the annotation. But at the moment, there's also no real need to delete it. I've used it for debugging and for reference, but I think cleaning it up is reasonable. Maybe as part of the action setting the final status in the CR. I'll take a look at it

[the remark this answers is not in the document]
Comment (donpenney) [reply adaptors/dell-hwmgr/nodepool.go:397]: I've created functions in the utils module for get/set/clear jobId to make things cleaner

**Pull request merged**